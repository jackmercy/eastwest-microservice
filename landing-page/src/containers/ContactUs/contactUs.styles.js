import { createUseStyles } from 'react-jss';

import { rem } from '../../utils/jss';
import COLORS from '../../constants/COLORS';
import BREAKPOINTS from '../../constants/BREAKPOINTS';

export default createUseStyles({
  teaser: {
    width: '100%',
    height: rem(350),
    boxSizing: 'border-box',
    backgroundImage: 'url(/static/images/contact/teaser.png)',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'right',
  },
  teaserContent: {
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    height: '100%',
    color: COLORS.textWhite,
  },
  teaserTitle: {
    margin: 0,
    fontWeight: 900,
    fontSize: rem(72),
    lineHeight: rem(88),
  },
  teaserDescription: {
    fontWeight: 600,
    fontSize: rem(24),
    lineHeight: rem(40),
    marginTop: rem(24),
    marginBottom: 0,
  },
  content: {
    display: 'flex',
    justifyContent: 'space-between',
    paddingTop: rem(64),
    paddingBottom: rem(64),
  },
  form: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
    width: '100%',
    maxWidth: rem(529),
    margin: rem(-8),
  },
  formInput: { width: '100%', padding: rem(8), boxSizing: 'border-box' },
  form30: { flex: '30%', maxWidth: '30%' },
  form50: { flex: '50%', maxWidth: '50%' },
  form70: { flex: '70%', maxWidth: '70%' },
  formInputPhoneLabel: { visibility: 'hidden' },
  formCheckBox: {
    display: 'flex',
    alignItems: 'center',
    width: '100%',
    maxWidth: `calc(100% - ${rem(16)})`,
    marginTop: rem(8),
  },
  formTerms: {
    fontWeight: 600,
    fontSize: rem(12),
    lineHeight: rem(16),
    color: COLORS.primary,
    cursor: 'pointer',
  },
  formButton: {
    width: '100%',
    maxWidth: `calc(100% - ${rem(16)})`,
    marginTop: rem(32),
  },
  contact: { maxWidth: rem(596), marginLeft: rem(16), overflow: 'hidden' },
  contactImage: {
    margin: [0, 'auto', rem(40)],
    maxWidth: rem(435),
    display: 'block',
    width: '100%',
  },
  contactLayout: {
    display: 'flex',
    alignItems: 'flex-end',
    margin: [0, rem(-16)],
    '& + $contactLayout': { marginTop: rem(24) },
  },
  contact50: { padding: [0, rem(16)], flex: '50%', maxWidth: '50%' },
  contact100: { padding: [0, rem(16)], flex: '100%', maxWidth: '100%' },
  contactRegion: {
    fontWeight: 900,
    fontSize: rem(24),
    lineHeight: rem(40),
    marginTop: 0,
    marginBottom: rem(12),
  },
  contactLabel: {
    fontWeight: 900,
    fontSize: rem(18),
    lineHeight: rem(32),
    marginTop: 0,
    marginBottom: rem(8),
  },
  contactText: {
    fontSize: rem(18),
    lineHeight: rem(32),
    margin: 0,
    '& > [class^=icon-]': { fontSize: rem(18), marginRight: rem(16) },
    '& + $contactText': { marginTop: rem(8) },
  },
  contactPhone: {
    color: COLORS.textBlack01,
    textDecoration: 'none',
    '&:hover': { color: COLORS.primary },
  },
  contactLink: {
    color: COLORS.primary,
    textDecoration: 'none',
  },
  modal: { maxWidth: rem(1440), top: '5%', maxHeight: '90%' },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxMd}px)`]: {
    content: {
      paddingTop: rem(25),
      flexDirection: 'column',
      alignItems: 'center',
    },
    contact: { marginLeft: 0, marginTop: rem(64) },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    teaser: {
      height: rem(248),
      backgroundImage: 'url(/static/images/contact/teaser-mobile.png)',
    },
    teaserTitle: { fontSize: rem(40), lineHeight: rem(48) },
    teaserDescription: {
      fontSize: rem(16),
      lineHeight: rem(24),
      marginTop: rem(16),
    },
    form: { maxWidth: '100%', margin: 0 },
    formInput: { padding: 0, '& + $formInput': { marginTop: rem(16) } },
    form30: { flex: '100%', maxWidth: '100%' },
    form50: { flex: '100%', maxWidth: '100%' },
    form70: { flex: '100%', maxWidth: '100%' },
    formInputPhoneLabel: { display: 'none' },
    formCheckBox: { maxWidth: '100%', marginTop: rem(16) },
    formButton: { maxWidth: '100%' },
    contact: { maxWidth: '100%' },
    contactImage: { marginBottom: rem(32) },
    contactLayout: { display: 'block', margin: 0 },
    contact50: {
      padding: 0,
      flex: '100%',
      maxWidth: '100%',
      '& + $contact50': { marginTop: rem(8) },
    },
    contact100: { padding: 0, flex: '100%', maxWidth: '100%' },
  },
});
