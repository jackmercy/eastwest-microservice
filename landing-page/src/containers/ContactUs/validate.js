import { formUtil } from '../../utils';

const validations = {
  reason: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
  firstName: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
  lastName: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
  phoneCode: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
  phoneNumber: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
  ],
  emailAddress: [
    { fn: formUtil.validation.require, error: 'FORM_VALIDATION.REQUIRED' },
    {
      fn: formUtil.validation.email,
      error: 'FORM_VALIDATION.EMAIL_INVALID',
    },
  ],
  agreeWithTerms: [{ fn: (value) => value, error: 'required' }],
};
const validate = (formValues) => formUtil.validate(formValues, validations);

export default validate;
