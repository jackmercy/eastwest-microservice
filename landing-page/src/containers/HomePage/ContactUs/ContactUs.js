import { memo } from 'react';
import { useIntl } from 'react-intl';
import Link from 'next/link';

import { Content } from '../../components';
import { Button } from '../../../components';

import useStyles from './contactUs.styles';

const ContactUs = () => {
  const intl = useIntl();
  const classes = useStyles();

  return (
    <div className={classes.root} data-test="contact-us-cover-image">
      <Content className={classes.content}>
        <h2 className={classes.title} data-test="contact-us-title">
          {intl.formatMessage({ id: 'HOMEPAGE.CONTACT_US.TITLE' })}
        </h2>
        <Link href="/contact" passHref>
          <Button
            tag="a"
            size="large"
            color="primary"
            keepSizeOnMobile
            className={classes.button}
            dataTest="contact-us-button"
          >
            {intl.formatMessage({ id: 'HOMEPAGE.CONTACT_US.BUTTON' })}
          </Button>
        </Link>
      </Content>
    </div>
  );
};

export default memo(ContactUs);
