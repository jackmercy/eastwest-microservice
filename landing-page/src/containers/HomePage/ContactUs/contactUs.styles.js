import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import COLORS from '../../../constants/COLORS';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    width: '100%',
    position: 'relative',
    boxSizing: 'border-box',
    height: rem(754),
    backgroundImage: 'url(/static/images/homepage/teaser-03.png)',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center bottom',
    transition: 'padding .5s ease',
    overflow: 'hidden',
  },
  content: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: rem(127),
    paddingBottom: rem(127),
  },
  title: {
    margin: 0,
    fontWeight: 900,
    fontSize: rem(100),
    lineHeight: rem(120),
    color: COLORS.textWhite,
    transition: 'font-size .5s ease',
  },
  button: {
    '&.large': {
      fontWeight: 600,
      fontSize: rem(32),
      lineHeight: rem(48),
      height: rem(64),
    },
    '&.primary': { color: COLORS.primary, backgroundColor: COLORS.bgWhite },
    padding: [0, rem(32)],
    marginLeft: rem(54),
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxLg}px)`]: {
    content: { display: 'block', textAlign: 'center' },
    title: { display: 'block' },
    button: { marginTop: rem(100), marginLeft: 0 },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: {
      backgroundImage: 'url(/static/images/homepage/teaser-03-mobile.png)',
      height: rem(520),
    },
    content: { paddingTop: rem(48), paddingBottom: rem(48) },
    title: { fontSize: rem(40), lineHeight: rem(48) },
    button: { marginTop: rem(100), marginLeft: 0 },
  },
});
