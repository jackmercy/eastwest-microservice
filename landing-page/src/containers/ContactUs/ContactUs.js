import { memo, useMemo } from 'react';
import { useIntl } from 'react-intl';
import classNames from 'classnames';
import Link from 'next/link';
import { useRouter } from 'next/router';
import csc from 'country-state-city';
import { useDispatch, useSelector } from 'react-redux';

import {
  SEO,
  Modal,
  ModalHeader,
  ModalContent,
  Input,
  Select,
  Button,
  Textarea,
  CheckBox,
  InputPhone,
} from '../../components';
import { stringUtil } from '../../utils';
import { Layout, Header, Footer, Content } from '../components';
import { PrivacyContent } from '../Privacy';
import { DEFAULT_SEO, LINKS } from '../../../configs';
import { useFormValidation, useModalState } from '../../hooks';
import { getLoading } from '../../redux/ducks/loading.duck';
import { callSendNewContactRequest } from '../../redux/ducks/form.duck';
import ModalSuccess from '../HomePage/CreateProject/ModalSuccess';
import { gaService } from '../../services';

import validate from './validate';
import useStyles from './contactUs.styles';

const ContactUs = () => {
  const intl = useIntl();
  const classes = useStyles();
  const dispatch = useDispatch();
  const router = useRouter();
  const isLoading = useSelector(getLoading);
  const [openModal, toggleModal] = useModalState(false);
  const [openSuccessModel, toggleSuccessModal] = useModalState(false);
  const reasonOptions = useMemo(
    () => [
      {
        label: intl.formatMessage({ id: 'CONTACT.FORM.REASON.OPTION.1' }),
        value: 1,
      },
      {
        label: intl.formatMessage({ id: 'CONTACT.FORM.REASON.OPTION.2' }),
        value: 2,
      },
      {
        label: intl.formatMessage({ id: 'CONTACT.FORM.REASON.OPTION.3' }),
        value: 3,
      },
      {
        label: intl.formatMessage({ id: 'CONTACT.FORM.REASON.OPTION.4' }),
        value: 4,
      },
      {
        label: intl.formatMessage({ id: 'CONTACT.FORM.REASON.OPTION.5' }),
        value: 5,
      },
      {
        label: intl.formatMessage({ id: 'CONTACT.FORM.REASON.OPTION.6' }),
        value: 6,
      },
      {
        label: intl.formatMessage({ id: 'CONTACT.FORM.REASON.OPTION.7' }),
        value: 7,
      },
    ],
    [intl],
  );
  const phoneCodeOptions = useMemo(
    () =>
      csc.getAllCountries().map((item) => ({
        label: `${item.name} (+${item.phonecode})`,
        value: item.isoCode,
        code: item.phonecode,
        default: item.isoCode === 'VN',
      })),
    [],
  );
  const countryOptions = useMemo(
    () =>
      csc
        .getAllCountries()
        .map((item) => ({ label: item.name, value: item.isoCode })),
    [],
  );
  const {
    errors,
    formValues,
    handleResetFieldError,
    handleSetMultipleFieldValues,
    handleSetFieldValue,
    handleResetFormValues,
    translateError,
    handleSubmit,
  } = useFormValidation(
    {
      address: '',
      lastName: '',
      firstName: '',
      phoneNumber: '',
      emailAddress: '',
      description: '',
      country: null,
      city: null,
      reason: router.query?.defaultReason
        ? reasonOptions.find(
          (r) => r.value === Number(router.query.defaultReason),
        )
        : null,
      agreeWithTerms: false,
      phoneCode: phoneCodeOptions.find((item) => item.default),
    },
    validate,
  );
  const cityOptions = useMemo(
    () =>
      (formValues.country
        ? csc
          .getStatesOfCountry(formValues.country.value)
          .map((item) => ({ label: item.name, value: item.isoCode }))
        : []),
    [formValues],
  );
  const formConfigs = useMemo(
    () => [
      {
        component: Select,
        options: reasonOptions,
        name: 'reason',
        label: intl.formatMessage({ id: 'CONTACT.FORM.REASON.LABEL' }),
        placeholder: intl.formatMessage({
          id: 'CONTACT.FORM.REASON.PLACEHOLDER',
        }),
        dataTest: 'contact-form-select-reason',
        required: true,
      },
      {
        component: Input,
        type: 'text',
        name: 'firstName',
        label: intl.formatMessage({ id: 'CONTACT.FORM.FIRST_NAME.LABEL' }),
        placeholder: intl.formatMessage({
          id: 'CONTACT.FORM.FIRST_NAME.PLACEHOLDER',
        }),
        className: classes.form50,
        dataTest: 'contact-form-input-first-name',
        required: true,
      },
      {
        component: Input,
        type: 'text',
        name: 'lastName',
        label: intl.formatMessage({ id: 'CONTACT.FORM.LAST_NAME.LABEL' }),
        placeholder: intl.formatMessage({
          id: 'CONTACT.FORM.LAST_NAME.PLACEHOLDER',
        }),
        className: classes.form50,
        dataTest: 'contact-form-input-last-name',
        required: true,
      },
      {
        component: Input,
        type: 'email',
        name: 'emailAddress',
        label: intl.formatMessage({ id: 'CONTACT.FORM.EMAIL_ADDRESS.LABEL' }),
        placeholder: intl.formatMessage({
          id: 'CONTACT.FORM.EMAIL_ADDRESS.PLACEHOLDER',
        }),
        dataTest: 'contact-form-input-email-address',
        required: true,
      },
      {
        component: Select,
        options: phoneCodeOptions,
        name: 'phoneCode',
        label: intl.formatMessage({ id: 'CONTACT.FORM.PHONE_NUMBER.LABEL' }),
        className: classes.form30,
        isSearchable: true,
        dataTest: 'contact-form-select-phone-code',
        required: true,
      },
      {
        component: InputPhone,
        name: 'phoneNumber',
        label: intl.formatMessage({ id: 'CONTACT.FORM.PHONE_NUMBER.LABEL' }),
        placeholder: intl.formatMessage({
          id: 'CONTACT.FORM.PHONE_NUMBER.PLACEHOLDER',
        }),
        className: classes.form70,
        labelClassName: classes.formInputPhoneLabel,
        dataTest: 'contact-form-input-phone-number',
        required: true,
      },
      {
        component: Input,
        type: 'text',
        name: 'address',
        label: intl.formatMessage({ id: 'CONTACT.FORM.ADDRESS.LABEL' }),
        placeholder: intl.formatMessage({
          id: 'CONTACT.FORM.ADDRESS.PLACEHOLDER',
        }),
        dataTest: 'contact-form-input-address',
      },
      {
        component: Select,
        options: countryOptions,
        name: 'country',
        label: intl.formatMessage({ id: 'CONTACT.FORM.COUNTRY.LABEL' }),
        placeholder: intl.formatMessage({
          id: 'CONTACT.FORM.COUNTRY.PLACEHOLDER',
        }),
        isSearchable: true,
        dataTest: 'contact-form-select-country',
      },
      {
        component: Select,
        options: cityOptions,
        name: 'city',
        label: intl.formatMessage({ id: 'CONTACT.FORM.CITY.LABEL' }),
        placeholder: intl.formatMessage({
          id: 'CONTACT.FORM.CITY.PLACEHOLDER',
        }),
        isSearchable: true,
        noOptionsMessage: () =>
          intl.formatMessage({ id: 'CONTACT.FORM.CITY.EMPTY' }),
        dataTest: 'contact-form-select-city',
      },
      {
        component: Textarea,
        name: 'description',
        label: intl.formatMessage({ id: 'CONTACT.FORM.DESCRIPTION.LABEL' }),
        placeholder: intl.formatMessage({
          id: 'CONTACT.FORM.DESCRIPTION.PLACEHOLDER',
        }),
        dataTest: 'contact-form-textarea-description',
      },
    ],
    [
      intl,
      classes,
      reasonOptions,
      countryOptions,
      cityOptions,
      phoneCodeOptions,
    ],
  );

  const handleSendEmail = () => {
    handleSubmit((formValues) => {
      dispatch(
        callSendNewContactRequest({
          formValues: {
            reason: formValues.reason.label,
            firstName: formValues.firstName,
            lastName: formValues.lastName,
            emailAddress: formValues.emailAddress,
            // eslint-disable-next-line max-len
            phoneNumber: `(+${formValues.phoneCode.code}) ${formValues.phoneNumber}`,
            address: formValues.address,
            country: formValues.country.label,
            city: formValues.city.label,
            description: formValues.description,
          },
          callback: (success) => {
            if (success) {
              toggleSuccessModal();
              gaService.event({
                action: 'submit',
                params: { form: 'contact', formValues },
              });
            }
          },
        }),
      );
    });
  };

  const handleFieldChange = ({ target, name, value }) => {
    if (name === 'country' && formValues.city) {
      handleSetMultipleFieldValues({
        [name]: value,
        city: null,
      });

      return;
    }
    if (target) {
      handleSetFieldValue(target.name, target.value);
    } else {
      handleSetFieldValue(name, value);
    }
  };

  const handleCloseSuccessModal = () => {
    handleResetFormValues();
    toggleSuccessModal();
  };

  return (
    <Layout>
      <SEO {...DEFAULT_SEO.PAGES.CONTACT} />
      <Header />
      <div className={classes.teaser} data-test="contact-teaser-image">
        <Content className={classes.teaserContent}>
          <h1 className={classes.teaserTitle} data-test="contact-teaser-title">
            {intl.formatMessage({ id: 'CONTACT.TEASER.TITLE' })}
          </h1>
          <p
            className={classes.teaserDescription}
            data-test="contact-teaser-description"
          >
            {intl.formatMessage({ id: 'CONTACT.TEASER.DESCRIPTION' })}
          </p>
        </Content>
      </div>
      <Content id="contact-form" className={classes.content}>
        <div className={classes.form}>
          {formConfigs.map(({ component: Component, className, ...rest }) => (
            <Component
              size="large"
              key={rest.name}
              onChange={handleFieldChange}
              onFocus={() => handleResetFieldError(rest.name)}
              error={translateError(errors[rest.name])}
              disabled={isLoading}
              className={classNames(classes.formInput, className)}
              value={formValues[rest.name]}
              {...rest}
            />
          ))}
          <div className={classes.formCheckBox}>
            <CheckBox
              name="agreeWithTerms"
              label={intl.formatMessage({
                id: 'CONTACT.FORM.AGREE_WITH_TERMS',
              })}
              defaultChecked={formValues.agreeWithTerms}
              onChange={({ target: { name, checked } }) => {
                handleSetFieldValue(name, checked);
              }}
              dataTest="contact-form-checkbox-privacy"
            />
            &nbsp;
            <span
              role="presentation"
              className={classes.formTerms}
              onClick={toggleModal}
            >
              {intl.formatMessage({
                id: 'CONTACT.FORM.AGREE_WITH_TERMS.PRIVACY_POLICY',
              })}
            </span>
          </div>
          <Button
            size="large"
            color="primary"
            disabled={!formValues.agreeWithTerms || isLoading}
            className={classes.formButton}
            onClick={handleSendEmail}
            dataTest="contact-form-button-submit"
          >
            {intl.formatMessage({ id: 'CONTACT.FORM.BUTTON.SUBMIT' })}
          </Button>
        </div>
        <div className={classes.contact}>
          <img
            alt="cover"
            loading="lazy"
            src="/static/images/contact/cover.png"
            srcSet={[
              '/static/images/contact/cover.png 500w',
              '/static/images/contact/cover@2x.png 1500w',
              '/static/images/contact/cover@3x.png 2000w',
            ].join(', ')}
            className={classes.contactImage}
            data-test="contact-info-image"
          />
          <div className={classes.contactLayout}>
            <div className={classes.contact50}>
              <h2 className={classes.contactRegion}>
                {intl.formatMessage({ id: 'CONTACT.INFO.REGION.1' })}
              </h2>
              <h3 className={classes.contactLabel}>
                {intl.formatMessage({ id: 'CONTACT.INFO.ADDRESS' })}
              </h3>
              <p
                className={classes.contactText}
                data-test="contact-info-ireland-address"
              >
                {intl.formatMessage({ id: 'CONTACT.INFO.ADDRESS.1' })}
              </p>
            </div>
            <div className={classes.contact50}>
              <h2 className={classes.contactRegion}>
                {intl.formatMessage({ id: 'CONTACT.INFO.REGION.2' })}
              </h2>
              <h3 className={classes.contactLabel}>
                {intl.formatMessage({ id: 'CONTACT.INFO.ADDRESS' })}
              </h3>
              <p
                className={classes.contactText}
                data-test="contact-info-vietnam-address"
              >
                {intl.formatMessage({ id: 'CONTACT.INFO.ADDRESS.2' })}
              </p>
            </div>
          </div>
          <div className={classes.contactLayout}>
            <div className={classes.contact50}>
              <h2 className={classes.contactLabel}>
                {intl.formatMessage({ id: 'CONTACT.INFO.PHONE' })}
              </h2>
              <a
                href={`tel:${stringUtil.removeSpaces(DEFAULT_SEO.PHONENUMBER)}`}
                className={classNames(
                  classes.contactText,
                  classes.contactPhone,
                )}
                data-test="contact-info-phone-number"
              >
                <i className="icon-phone" />
                {DEFAULT_SEO.PHONENUMBER}
              </a>
            </div>
            <div className={classes.contact50}>
              <Link href="/create-a-ticket" passHref>
                <a
                  href="/#"
                  className={classNames(
                    classes.contactText,
                    classes.contactLink,
                  )}
                  data-test="contact-info-cta-create-ticket"
                >
                  {intl.formatMessage({ id: 'CONTACT.INFO.CREATE_TICKET' })}
                </a>
              </Link>
            </div>
          </div>
          <div className={classes.contactLayout}>
            <div className={classes.contact100}>
              <h2 className={classes.contactLabel}>
                {intl.formatMessage({ id: 'CONTACT.INFO.SERVICES' })}
              </h2>
              <p
                className={classes.contactText}
                data-test="contact-info-link-messenger"
              >
                <i className="icon-messenger" />
                <a
                  href={LINKS.SOCIALS.MESSENGER}
                  title={LINKS.SOCIALS.MESSENGER}
                  target="_blank"
                  rel="noreferrer"
                  className={classes.contactLink}
                >
                  {LINKS.SOCIALS.MESSENGER}
                </a>
              </p>
              <p
                className={classes.contactText}
                data-test="contact-info-link-linkedin"
              >
                <i className="icon-linkedin-white" />
                <a
                  href={LINKS.SOCIALS.LINKEDIN}
                  title={LINKS.SOCIALS.LINKEDIN}
                  target="_blank"
                  rel="noreferrer"
                  className={classes.contactLink}
                >
                  {LINKS.SOCIALS.LINKEDIN}
                </a>
              </p>
            </div>
          </div>
        </div>
      </Content>
      <Footer />
      <Modal open={openModal} toggle={toggleModal} className={classes.modal}>
        <ModalHeader toggle={toggleModal} title="Privacy policy" />
        <ModalContent>
          <PrivacyContent />
        </ModalContent>
      </Modal>
      <ModalSuccess
        open={openSuccessModel}
        toggle={handleCloseSuccessModal}
        name={`${formValues.firstName} ${formValues.lastName}`}
      />
    </Layout>
  );
};

export default memo(ContactUs);
