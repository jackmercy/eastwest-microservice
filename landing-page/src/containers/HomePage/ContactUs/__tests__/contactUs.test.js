import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ContactUs from '../ContactUs';

describe('Testing HomePage > ContactUs', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<ContactUs />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<ContactUs />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a cover image', () => {
    expect(wrapper.find('[data-test="contact-us-cover-image"]')).toHaveLength(
      1,
    );
  });

  it('should have a title', () => {
    expect(wrapper.find('[data-test="contact-us-title"]')).toHaveLength(
      1,
    );
  });

  it('should have a CTA button', () => {
    expect(wrapper.find('[data-test="contact-us-button"]')).toHaveLength(1);
  });
});
