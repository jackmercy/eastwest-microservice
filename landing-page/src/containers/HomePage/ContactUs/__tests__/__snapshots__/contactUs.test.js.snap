// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomePage > ContactUs should render match snapshot 1`] = `
<div
  className="root-0-2-1"
  data-test="contact-us-cover-image"
>
  <Memo(Content)
    className="content-0-2-2"
  >
    <h2
      className="title-0-2-3"
      data-test="contact-us-title"
    >
      Let talk, why not?
    </h2>
    <Link
      href="/contact"
      passHref={true}
    >
      <Memo([object Object])
        className="button-0-2-4"
        color="primary"
        dataTest="contact-us-button"
        keepSizeOnMobile={true}
        size="large"
        tag="a"
      >
        Drop us a line
      </Memo([object Object])>
    </Link>
  </Memo(Content)>
</div>
`;
